stable_baselines3/common/buffers.py

```diff
@@ -132,8 +132,10 @@ def to_torch(self, array: np.ndarray, copy: bool = True) -> th.Tensor:
         :return:
         """
         if copy:
-            return th.tensor(array).to(self.device)
-        return th.as_tensor(array).to(self.device)
+            return th.tensor(array, device=self.device)
+        if self.device != th.device("cpu"):
```

Comment:
I don't think we need the warning there, I would put it in the doc as it is for `as_tensor`: https://pytorch.org/docs/stable/generated/torch.as_tensor.html

"If data is already a tensor with the requested dtype and device then data itself is returned, but if data is a tensor with a different dtype or device then it’s copied as if using data.to(dtype=dtype, device=device)."

Reply:
ok for me